internal/proxy/docker/docker_test.go

```diff
@@ -687,6 +687,88 @@ func TestAddBinding(t *testing.T) {
 	}
 }
 
+func TestGetNetwork(t *testing.T) {
+	testCases := []struct {
+		name        string
+		imageSpec   []byte
+		wantNetwork string
+		wantErr     bool
+	}{
+		{
+			name:        "Valid network mode",
+			imageSpec:   []byte(`{"HostConfig": {"NetworkMode": "bridge"}}`),
+			wantNetwork: "bridge",
+			wantErr:     false,
+		},
+		{
+			name:        "Missing NetworkMode",
+			imageSpec:   []byte(`{"HostConfig": {}}`),
+			wantNetwork: "",
+			wantErr:     true,
+		},
+		{
+			name:        "Invalid JSON",
+			imageSpec:   []byte(`{"HostConfig": {`),
+			wantNetwork: "",
+			wantErr:     true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			gotNetwork, err := getNetwork(tc.imageSpec)
+			if (err != nil) != tc.wantErr {
+				t.Errorf("getNetwork() error = %v, wantErr %v", err, tc.wantErr)
+				return
+			}
+			if gotNetwork != tc.wantNetwork {
+				t.Errorf("getNetwork() = %v, want %v", gotNetwork, tc.wantNetwork)
+			}
+		})
+	}
+}
+
+func TestSetNetwork(t *testing.T) {
+	testCases := []struct {
+		name       string
+		imageSpec  []byte
+		newNetwork string
+		wantSpec   []byte
+		wantErr    bool
+	}{
+		{
+			name:       "Set new network",
```

Comment:
It also might be worth testing when a network wasn't provided in the input (like getNetwork, but I think this time it's not an error)?